rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/client/LRAIT.java

```diff
@@ -0,0 +1,192 @@
+import org.eclipse.microprofile.lra.annotation.LRAStatus;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import static org.eclipse.microprofile.lra.annotation.ws.rs.LRA.LRA_HTTP_CONTEXT_HEADER;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Arquillian.class)
+public class LRAIT {
+
+    @ArquillianResource
+    private URL baseURL;
+
+    private Client client;
+
+    @Deployment
+    public static WebArchive deploy() {
+        return Deployer.deploy(LRAIT.class.getSimpleName());
+    }
+
+    @Before
+    public void before() {
+        client = ClientBuilder.newClient();
+    }
+
+    @After
+    public void after() {
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    /**
+     * Invoke a resource method which in turn invokes other resources.
+     * The various resource invocations are called in new or existing LRAs.
+     * Various tests are performed to verify that the correct LRAs are used
+     * and the LRAs have the expected status (see the resource method for
+     * the detail).
+     *
+     * @throws URISyntaxException
+     */
+    @Test
+    public void testChainOfInvocations() throws URISyntaxException {
+        NarayanaLRAClient lraClient = new NarayanaLRAClient();
+
+        // Invoke a method which starts a transaction
+        // (note that the method LRAParticipant.CREATE_OR_CONTINUE_LRA also invokes other resource methods)
+        URI lra1 = invokeInTransaction(null, LRAParticipant.RESOURCE_PATH, LRAParticipant.CREATE_OR_CONTINUE_LRA, 200);
+        assertEquals("lra1 should be active", LRAStatus.Active, lraClient.getStatus(lra1));
+
+        // end the LRA
+        invokeInTransaction(lra1, LRAParticipant.RESOURCE_PATH, LRAParticipant.END_EXISTING_LRA, 200);
+
+        assertNull("remote invocations should not update current (see MP-LRA issue behaviour)", lraClient.getCurrent());
+    }
+
+    /**
+     * test behaviour when multiple LRAs are active
+     */
+    @Test
+    public void testNoCurrent() throws URISyntaxException {
+        NarayanaLRAClient lraClient = new NarayanaLRAClient();
+        URI lra1 = invokeInTransaction(null, LRAParticipant.RESOURCE_PATH, LRAParticipant.START_NEW_LRA, 200);
+        assertNull("remote invocations should not update current (see MP-LRA issue behaviour)", lraClient.getCurrent());
```

Comment:
does make sense to add here some more context about the failure? What does the `current` mean - is it currently active LRA instance? I  think it could be hard to search expected behaviour in MP-LRA with this description.